src/network/serverpackethandler.cpp

```diff
@@ -1154,8 +1154,11 @@ void Server::handleCommand_Interact(NetworkPacket *pkt)
 			u16 src_original_hp = pointed_object->getHP();
 			u16 dst_origin_hp = playersao->getHP();
 
-			pointed_object->punch(dir, &toolcap, playersao,
+			s16 wear = pointed_object->punch(dir, &toolcap, playersao,
 					time_from_last_punch);
+			bool result = punchitem.addWear(wear, m_itemdef);
+			if (result && playersao->setWieldedItem(punchitem))
+				SendInventory(playersao);
```

Comment:
This function has been recently modified to take two params, the second one being a boolean that determines if it's enough to only send the changed portion of the inventory. AFAIK adding `true` ought to fix it.

```suggestion
				SendInventory(playersao, true);
```

It'd be nice if @SmallJoker could confirm this.